src/Common/src/System/Net/Http/HttpHandlerDiagnosticListenerExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Defines default values for http handler properties which is meant to be re-used across WinHttp and UnixHttp Handlers
+    /// </summary>
+    internal static class HttpHandlerDiagnosticListenerExtensions
+    {
+        public static Guid LogHttpRequest(this DiagnosticListener @this, HttpRequestMessage request)
+        {
+            if (@this.IsEnabled(HttpHandlerLoggingStrings.RequestWriteName))
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                long timestamp = Stopwatch.GetTimestamp();
+
+                @this.Write(
+                    HttpHandlerLoggingStrings.RequestWriteName,
+                    new
+                    {
+                        Request = request,
+                        LoggingRequestId = loggingRequestId,
+                        Timestamp = timestamp
+                    }
+                );
+
+                return loggingRequestId;
+            }
+
+            return Guid.Empty;
+        }
```

Comment:
Regarding my comment about inlining, from a perf perspective it may of course not matter, but my intention was that this be changed to something like:
```C#
public static Guid LogHttpRequest(this DiagnosticListener @this, HttpRequestMessage request)
{
    return @this.IsEnabled(HttpHandlerLoggingStrings.RequestWriteName) ?
        LogHttpRequestCore(@this, request) :
        Guid.Empty;
}
```
and then having all of the work in a private method that's delegated to.  I'm fine with you going ahead without that, as given everything else that's going on, it's unlikely to make a difference.  We can measure it after (for reference, you can use PerfView to see whether inlining is happening or not; it has a JIT Inlining feature that'll show you everything that's inlined, everything that's not, and why).